File: Week-1/Mini-MaxSum/src/main/java/MyCode.java
import java.util.Collection;
import java.util.Collections;
import java.util.List;
import java.util.stream.Collectors;

public class MyCode {
    public static void miniMaxSum(List<Integer> arr) {
        Collections.sort(arr);
        List<Long> arrLong = arr.stream().map(Integer::longValue).collect(Collectors.toList());
        // arrLong.forEach(System.out::println);
        Long maxSum = (arrLong.stream().reduce(1L, (a,b) -> a+b )) - arrLong.get(0);
        Long minSum = (arrLong.stream().reduce(1L, (a,b) -> a+b )) - arrLong.get(arrLong.size()-1);
        System.out.println(maxSum + " " + minSum);
    }

}

File: Week-1/Mini-MaxSum/src/main/kotlin/MyCode.kt

fun miniMaxSum(arr: Array<Int>): Unit {
    var minSum : Long = 0L
    var maxSum : Long = 0L
    arr.sort()
    if (arr.first() == arr.last()){
        println("${arr.sum() - arr[0]} ${arr.sum() - arr[0]}")
    }else{
        for(i in arr){
            if( i > arr.first()) maxSum += i
            if( i<arr.last()) minSum += i
        }
        println("$minSum $maxSum")
    }

}

fun main(args: Array<String>) {

    val arr = readLine()!!.trimEnd().split(" ").map{ it.toInt() }.toTypedArray()

    miniMaxSum(arr)
}